GoogleUtilities/Environment/GULHeartbeatDateStorage.m

```diff
@@ -32,11 +36,21 @@ - (instancetype)initWithFileName:(NSString *)fileName {
   self = [super init];
   if (self) {
     _fileCoordinator = [[NSFileCoordinator alloc] initWithFilePresenter:nil];
+    _fileName = fileName;
+  }
+  return self;
+}
+
+/** Lazy getter for fileURL
+ * @return fileURL where heartbeat information is stored.
+ */
+- (NSURL *)fileURL {
+  if (!_fileURL) {
     NSURL *directoryURL = [[self class] directoryPathURL];
     [[self class] checkAndCreateDirectory:directoryURL fileCoordinator:_fileCoordinator];
-    _fileURL = [directoryURL URLByAppendingPathComponent:fileName];
+    _fileURL = [directoryURL URLByAppendingPathComponent:_fileName];
   }
-  return self;
+  return _fileURL;
 }
```

Comment:
I went back and forth with what would be an ideal way to instantiate the `fileURL` property.  What do you think @maksymmalyhin ?